specs/experimental/custom-gas-token.md

```diff
@@ -0,0 +1,306 @@
+- `data`: TODO
+- `sourceHash`: TODO
+
+### L2StandardBridge Deployment
+
+- `from`: `0x4210000000000000000000000000000000000004`
+- `to`: `null`
+- `mint`: `0`
+- `value`: `0`
+- `gasLimit`: `375,000`
+- `data`: TODO
+- `sourceHash`: TODO
+
+### L1Block Proxy Update
+
+- `from`: `0x0000000000000000000000000000000000000000`
+- `to`: `0x4200000000000000000000000000000000000015`
+- `mint`: `0`
+- `value`: `0`
+- `gasLimit`: `50,000`
+- `data`: TODO
+- `sourceHash`: TODO
+
+### L2CrossDomainMessenger Proxy Update
+
+- `from`: `0x0000000000000000000000000000000000000000`
+- `to`: `0x4200000000000000000000000000000000000007`
+- `mint`: `0`
+- `value`: `0`
+- `gasLimit`: `50,000`
+- `data`: TODO
+- `sourceHash`: TODO
+
+### L2StandardBridge Proxy Update
+
+- `from`: `0x0000000000000000000000000000000000000000`
+- `to`: `0x4200000000000000000000000000000000000010`
+- `mint`: `0`
+- `value`: `0`
+- `gasLimit`: `50,000`
+- `data`: TODO
+- `sourceHash`: TODO
+
+## Selection of `ETHER_TOKEN_ADDRESS`
+
+It was decided to use `address(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)` to represent `ether` to push ecosystem standardization
+efforts around using this address to represent `ether` in [DeFi protocols](https://twitter.com/transmissions11/status/1775525755860169010)
+or [APIs](https://github.com/ethereum/execution-apis/pull/484).
+
+## Security Considerations
+
+### OptimismPortal Token Allowance
+
+The `OptimismPortal` makes calls on behalf of users. It is therefore unsafe to be able to call the address of the custom gas
+token itself from the `OptimismPortal` because it would be a simple way to `approve` an attacker's balance and steal the
+entire ERC20 token balance of the `OptimismPortal`.
+
+### Decimal Scaling
+
+A library for scaling `uint256` representations of floating point numbers is used on both ERC20 token deposits and withdrawals.
+This ensures that from the point of view of the L2 system, it always appears as if the number of tokens that were deposited
+are the same as the native balance. Tokens with less than or 18 decimals are supported, meaning that there is no potential for
+loss of precision due to sending in too small of a unit.
+
+### Interoperability Support
+
+Interop is supported between chains that use a custom gas token. The token address and the number of decimals are legible on
+chain. In the future we may add the ability to poke a chain such that it emits an event that includes the custom gas token address
+and its number of decimals to easily be able to introspect on the native asset of another chain.
+
+### Wrapped Ether
+
+The `WETH9` predeploy at `0x4200000000000000000000000000000000000006` represents wrapped native asset and not wrapped `ether`.
+Portable and fungible `ether` across different domains is left for a future project.
+
+## Future Directions
+
+### Enabling a L2 Native ERC20 Token to Pay for Gas
+
+This is an experimental feature and it is not clear if it will ever be included in the OP Stack.
```

Comment:
In what way is it experimental? I interpreted this section to mean it's a work in progress and may never be completed. If that is correct, what do you think about removing this from the spec and moving this content to an issue? I believe so far the custom gas token demand is all for the L1-based token that's spec'd out, so we could leave this out to keep it more focused, and if we do hear demand for this feature and plan to implement we can add it back to the spec to collect feedback at that time.

Reply:
Ok good thinking, I think i just got a little trigger happy when working on this. Will move to an issue

Reply:
Removed mentions, tracking in https://github.com/ethereum-optimism/specs/issues/114